internal/fields/validate.go

```diff
@@ -212,7 +211,17 @@ func parseElementValue(key string, definition FieldDefinition, val interface{})
 
 	var valid bool
 	switch definition.Type {
-	case "date", "ip", "constant_keyword", "keyword", "text":
+        case "constant_keyword", "keyword":
+                // Accept numeric fields as valid keywords.
+                if _, isStr := val.(string); !isStr {
```

Comment:
It looks like the PR is an answer to the https://github.com/elastic/integrations/issues/422 . Actually I'm a fan of the stricter validation, but I'm fine with adjusting the tool if necessary.

Any thoughts about this PR? @ycombinator @andrewkroh

Reply:
This is the opposite case. 

https://github.com/elastic/integrations/issues/422 wants to ingest strings into numeric fields, which is dangerous because you'll get an exception if ES can't convert the string to the number type (or if coercion is disabled). It's better to validate and convert those fields in a pipeline.

What this PR does is allow numeric values to be ingested as keywords, which is a common pattern for any field that is numeric but has a symbolic meaning and doesn't require `range` queries. In this case the only risk is that if no mapping is present in the index, ES will create the field as `long` instead of `keyword`. But I don't think that's possible with ingest manager.

Reply:
Thank you for the explanation, @adriansr. It would be great to add a similar justifying comment in the code. As it's a non-intuitive conversion between fields, let's a unit test around this function (`parseElementValue`).

I wonder if we should describe it in the docs/howto guides, but I leave it up to you to decide if it's useful or not.